src/components/scenes/existingAccout/ChangeRecoveryScene.tsx

```diff
@@ -1,523 +1,505 @@
+  const handleQuestion2 = () => {
+    Airship.show<string | undefined>(bridge => (
+      <RadioListModal
+        bridge={bridge}
+        title={s.strings.recovery_question_two}
+        items={question2pool}
+        selected={question2}
+      />
+    )).then(question2 => {
+      if (question2 == null) return
+      setQuestion2(question2)
+      setAnswer2('')
     })
   }
 
-  handleQuestionSelected = (data: EdgeRecoveryQuestionChoice) => {
-    const question = data.question
-    if (this.state.focusFirst) {
-      this.setState({
-        question1: question,
-        showQuestionPicker: false
-      })
-      return
-    }
-    this.setState({
-      question2: question,
-      showQuestionPicker: false
+  const showDatePickerModal = (
+    setDate: (date: string) => void,
+    setError: (error: string | null) => void
+  ) => {
+    Airship.show<Date | undefined>(bridge => (
+      <DateModal bridge={bridge} />
+    )).then(answer => {
+      if (answer == null) return
+      setDate(toLocalDate(answer))
+      setError(null)
     })
   }
 
-  async enableRecovery(): Promise<void> {
-    const buttons: { [key: string]: ButtonInfo } = {
-      email: {
-        label: s.strings.confirm_email,
-        onPress: this.handleEnableEmail
-      },
-      share: {
-        label: s.strings.confirm_share,
-        onPress: this.handleEnableShare
-      },
-      cancel: { label: s.strings.cancel, type: 'secondary' }
+  const showTextInputModal = async (
+    title: string,
+    question: EdgeRecoveryQuestionChoice,
+    setText: (answer: string) => void,
+    setError: (error: string | null) => void
+  ) => {
+    const onSubmit = async (answer: string): Promise<true | string> => {
+      const ok = answer.length >= question.min_length
+      const errorMsg = sprintf(s.strings.min_length_error, question.min_length)
+      if (ok) setText(answer)
+      else setError(errorMsg)
+      return ok ? true : errorMsg
     }
-    if (Platform.OS === 'android') delete buttons.email
-
-    // Ask which way to send the key:
-    await Airship.show(bridge => (
-      <ButtonsModal
+    await Airship.show<string | undefined>(bridge => (
+      <TextInputModal
         bridge={bridge}
-        title={s.strings.confirm_recovery_questions}
-        buttons={buttons}
-      >
-        <MessageText>{this.state.question1}</MessageText>
-        <MessageText>
-          <Strong>{this.state.answer1}</Strong>
-        </MessageText>
-        <MessageText>{this.state.question2}</MessageText>
-        <MessageText>
-          <Strong>{this.state.answer2}</Strong>
-        </MessageText>
-      </ButtonsModal>
+        title={title}
+        inputLabel={s.strings.recovery_answer_placeholder}
+        returnKeyType="go"
+        autoCapitalize="none"
+        autoCorrect={false}
+        autoFocus
+        onSubmit={onSubmit}
+      />
     ))
   }
+  const handleAnswer1 = () => {
+    if (edgeQuestion1 == null) return
+    edgeQuestion1.category === 'numeric'
```

Comment:
Cannot enforce a data picker for existing old questions. They may have previously been answered in an unknown format ie "Jan 1, 2001" or "1/1/1".  Must create new questions with slightly different test and remove the old questions. Make the category `date`. If a user is actually doing the recovery process, you must properly handle unknown questions and provide a generic text input.

Let's sidebar on this as this is a critical bug that needs to be handled correctly.

Reply:
Legacy case handled in https://github.com/EdgeApp/edge-login-ui-rn/pull/21/commits/e249962b1d9dded1e593431e314581b217733c7b#diff-c70b5cca0121a203a94f0562ab9ac79e2eb43849b006178e6f3cc640cdee20baR88